pystapi-client/src/pystapi_client/exceptions.py

```diff
@@ -0,0 +1,17 @@
+from httpx import Response
+
+
+class APIError(Exception):
+    """Raised when unexpected server error."""
+
+    status_code: int | None
+
+    @classmethod
+    def from_response(cls, response: Response) -> "APIError":
```

Comment:
Let's do any `from __future__ import annotations` required to not have to quote the type.

```suggestion
    def from_response(cls, response: Response) -> APIError:
```